src/poetry_plugin_freeze/app.py

```diff
@@ -232,13 +232,15 @@ def get_frozen_deps(self, dep_packages, exclude_packages=None):
 
     def replace_deps(self, dist_meta, dep_lines):
         start_pos = 0
-        for m in dist_meta.get_all("Requires-Dist"):
-            if not start_pos:
-                start_pos = dist_meta._headers.index(("Requires-Dist", m))
-            dist_meta._headers.remove(("Requires-Dist", m))
-
-        for idx, h in enumerate(dep_lines):
-            dist_meta._headers.insert(start_pos + idx, ("Requires-Dist", h))
+        requires_dist = dist_meta.get_all("Requires-Dist")
+        if requires_dist:
+            for m in requires_dist:
+                if not start_pos:
+                    start_pos = dist_meta._headers.index(("Requires-Dist", m))
+                dist_meta._headers.remove(("Requires-Dist", m))
+
+            for idx, h in enumerate(dep_lines):
+                dist_meta._headers.insert(start_pos + idx, ("Requires-Dist", h))
```

Comment:
Good catch here! Though I think if we don't have any dependencies to freeze, we can probably avoid calling `replace_deps()` in the first place by making a change later in the file:

```diff
@@ -296,7 +295,8 @@ class IcedPoet:
             deps = self.get_path_deps(MAIN_GROUP)
             deps.update(dep_packages)
             dep_lines = self.get_frozen_deps(deps, self.exclude_packages)
-            self.replace_deps(dist_meta, dep_lines)
+            if dep_lines:
+                self.replace_deps(dist_meta, dep_lines)
 
             with source_whl.open(record_path) as record_fh:
                 record_text = self.freeze_record(record_fh, dist_meta, md_path)
```

It's still a good call to play it safe and avoid the chance for a `None` here, but we should still be able to cover that without indenting all of this logic:

```suggestion
        for m in dist_meta.get_all("Requires-Dist", ()):
            if not start_pos:
                start_pos = dist_meta._headers.index(("Requires-Dist", m))
            dist_meta._headers.remove(("Requires-Dist", m))

        for idx, h in enumerate(dep_lines):
            dist_meta._headers.insert(start_pos + idx, ("Requires-Dist", h))
```

Reply:
Good Morning and thanks for your quick response!
I absolutely agree with your suggestions, which is why i added them already into my PR.

> I'm inclined to say that it's not worth freezing a wheel with no dependencies, but that also shouldn't break.

Of course - i absolutely understand that point. But we are facing a edge case on our side - theres a pipeline template which is executing `poetry build` and `poetry freeze-wheel` for all of our python projects. Of course, we can put the logic also into the pipeline with some greping and if-conditions, which actually looked very hacky to us - thats the reason, why we decided to contribute the fix into this project. :)